app/services/current-session.js

```diff
@@ -38,21 +44,34 @@ export default class CurrentSessionService extends Service {
     this.user = null;
     this.role = null;
     this.organization = null;
+    this.impersonation.clear();
   }
 
   may(permission) {
     return this.userGroup?.permissions.includes(permission);
   }
 
+  reallyMay(permission) {
+    return this.ownUserGroup?.permissions.includes(permission);
+  }
```

Comment:
Isn't it more elegant to add an extra option to `may(permission)` which is `false` by default to validate on the impersonator role/group instead? Something like `may(permission, checkImpersonator = false)`